lib/src/fixed_priority_scheduler.rs

```diff
@@ -26,15 +30,27 @@ where
         Self {
             dag: dag.clone(),
             processor: processor.clone(),
+            node_log: vec![NodeLog::default(); dag.node_count()],
+            processor_log: ProcessorLog {
```

Comment:
It is more natural to implement the new method and assign the core_id here.

Reply:
Add `new()` function to each structure
In addition, the id of node and core are changed to be set to a value at the time of `new()`.